.github/actions/pre-commit/action.yml

```diff
@@ -17,10 +23,16 @@ runs:
   using: "composite"
   steps:
     - uses: actions/checkout@v3
+      env:
+        REF_TO_CHECKOUT: ${{ inputs.auto-commit == 'true' && github.head_ref || '' }}
+      with:
+        ref: ${{ env.REF_TO_CHECKOUT }}
+
```

Comment:
it's ugly to use an env but debugging is more easy

in this way we continue to checkout the merged head for PR when not using auto-commit